src/sentry/integrations/notifydisable.py

```diff
@@ -0,0 +1,60 @@
+from sentry import integrations
+from sentry.models import Organization
+from sentry.services.hybrid_cloud.integration import RpcIntegration
+from sentry.utils.email import MessageBuilder
+
+provider_types = {
+    "integration": "integrations",
+    "plugin": "plugins",
+    "sentry-app": "sentry-apps",
+}
+
+
+def get_url(organization: Organization, provider_type: str, provider: str) -> str:
+    if not provider_type:
+        return str(organization.absolute_url("/settings/"))
+
+    type_name = provider_types.get(provider_type, "")
+    return str(
+        organization.absolute_url(
+            f"/settings/{type_name}/{provider}/",
+        )
+    )
+
+
+def get_provider_type(redis_key: str) -> str:
+    for provider in provider_types:
+        if provider in redis_key:
+            return provider
+
+    return ""
+
+
+def get_subject(integration_name: str) -> str:
+    return f"Action required: re-authenticate or fix your {integration_name} integration"
+
+
+def notify_disable(
+    organization: Organization, integration: RpcIntegration, redis_key: str, project: None
```

Comment:
This should be `project: Project = None` (or `project: str = None` if it's the slug, I'm not sure) to indicate the type and that it's optional.